tests/unit/AmalgamatingBusinesses.spec.ts

```diff
@@ -87,15 +93,368 @@ describe.skip('Amalgamating Businesses - add amalgamating business', () => {
+    expect(store.stateModel.amalgamation.amalgamatingBusinesses.length).toBe(0)
+
+    // verify snackbar is displayed
+    expect(wrapper.vm.snackbar).toBe(true)
+    expect(wrapper.vm.snackbarText).toBe('Unable to add that business.')
+
+    vi.resetAllMocks()
+  })
+
+  it('doesn\'t save an amalgamating business - BC - missing first filing', async () => {
+    // open panel
+    await wrapper.setData({ isAddingAmalgamatingBusiness: true })
+
+    // mock services functions
+    vi.spyOn((AuthServices as any), 'fetchAuthInfo').mockImplementation(() => ({}))
+    vi.spyOn((LegalServices as any), 'fetchBusinessInfo').mockImplementation(() => ({}))
+    vi.spyOn((LegalServices as any), 'fetchAddresses').mockImplementation(() => ({}))
+    vi.spyOn((LegalServices as any), 'fetchFirstOrOnlyFiling').mockImplementation(() => null)
+
+    // simulate saving a BC business
+    await wrapper.vm.saveAmalgamatingBusiness({
+      legalType: CorpTypeCd.BC_COMPANY,
+      name: 'My BC Business',
+      identifier: 'BC1234567'
+    })
+
+    // verify data
+    expect(store.stateModel.amalgamation.amalgamatingBusinesses.length).toBe(0)
+
+    // verify snackbar is displayed
+    expect(wrapper.vm.snackbar).toBe(true)
+    expect(wrapper.vm.snackbarText).toBe('Unable to add that business.')
+
+    vi.resetAllMocks()
   })
 
-  it('saves an amalgamating business', async () => {
+  it('doesn\'t save a duplicate amalgamating business - Extra Pro', async () => {
+    // pre-populate an extra pro business
+    store.stateModel.amalgamation.amalgamatingBusinesses = [
+      {
+        type: AmlTypes.FOREIGN,
+        role: AmlRoles.AMALGAMATING,
+        foreignJurisdiction: { country: 'CA', region: 'British Columbia' },
+        legalName: 'Extra Pro Business',
+        corpNumber: 'A1234567'
+      }
+    ]
+    expect(store.stateModel.amalgamation.amalgamatingBusinesses.length).toBe(1)
+
+    // open panel
+    await wrapper.setData({ isAddingAmalgamatingBusiness: true })
+
+    // verify snackbar is not displayed
+    expect(wrapper.vm.snackbar).toBe(false)
+
+    // try to save a duplicate extra pro business
+    await wrapper.vm.saveAmalgamatingBusiness({
+      legalType: CorpTypeCd.EXTRA_PRO_A,
+      name: 'Extra Pro Business',
+      identifier: 'A1234567'
+    })
+
+    // verify snackbar is displayed
+    expect(wrapper.vm.snackbar).toBe(true)
+    expect(wrapper.vm.snackbarText).toBe('Business is already in table.')
+
+    // verify data
+    expect(store.stateModel.amalgamation.amalgamatingBusinesses.length).toBe(1)
+
+    // verify panel is now closed
+    expect(wrapper.vm.isAddingAmalgamatingBusiness).toBe(false)
   })
 
-  it('saves an amalgamating business', async () => {
+  it('save an amalgamating business - BC', async () => {
+    // set state
+    store.setCurrentDate('2023-01-01')
+    await Vue.nextTick()
+
+    // open panel
+    await wrapper.setData({ isAddingAmalgamatingBusiness: true })
+
+    // mock services functions
```

Comment:
At least look at this section 😆 

Spying on service functions and mocking their implementation (or result) is super handy.

Reply:
Good learning here!

Reply:
Love it.